File: intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> ans;
        map<int,int> res;
        for(auto &i :  nums2){
            if(!res[i]){
                res[i] = 1;
            }
            else{
                res[i] += 1 ;
            }
        }
        for(auto &i :nums1){
            auto it =  find(nums2.begin(),nums2.end(),i);
            if(it !=  nums2.end() ){ 
               if(res[i] > 0 ){
                ans.push_back(i);
                res[i] -=1;
            }
        }
        }
        return ans;
    }
};

File: intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1, nums2):
        z =[]
    
        
        for i in set(nums1):
              if i in nums2  :
                    x = min(nums1.count(i),nums2.count(i))
                    for j in range(x) :
                        z.append(i)
                    
        return z